sbt_common/internal_model/src/test/scala/uk/ac/wellcome/models/transformable/sierra/SierraRecordNumberTest.scala

```diff
@@ -0,0 +1,51 @@
+package uk.ac.wellcome.models.transformable.sierra
+
+import org.scalatest.{FunSpec, Matchers}
+import org.scalatest.prop.TableDrivenPropertyChecks._
+
+class SierraRecordNumberTest extends FunSpec with Matchers {
+
+  describe("withCheckDigit") {
+    val testCases = Table(
```

Comment:
Are these all different test cases that test different scenarios? If yes, I'd much prefer them to be separate test with individual titles. If not, we should delete the not useful ones. I think in these cases test titles are much better than comments to explain the reason for tests and are much more visible. Code duplication can be reused by using shared functions